### DISCUSSION TITLE
MINI example

### DISCUSSION THREAD

**Discussion by gdmcbain** [kinnala/scikit-fem#722] [Ideas]
The (triangular) MINI element `ElementTriMini` was introduced in #329 but without an example other than a couple of tests.  Should it be used in an example?

`ElementTetMini` is proposed in #371.

**Comment by kinnala** [comment]
A simple example would be to study the null space of the discrete Stokes operator using a few different elements, e.g.

1. P_1 - P_1 (unstable)
2. P_1 + bubble - P_1 (stable)
3. P_2 + P_1 (stable)

For unstable elements, the null space (eigenfunction corresponding to the eigenvalue 0) should contain a checkerboard function which can be visualized.

There is also some quadrilateral element which is stable only if the mesh is perfectly aligned to a grid but cannot remember which pair it is.

**Comment by gdmcbain** [comment]
I like this idea a lot.

I've just acquired a copy of

* Boffi, D., Brezzi, F. & Fortin, M. (2008). Finite elements for the Stokes problem. In Boffi & Gastaldi (eds) _Mixed Finite Elements, Compatibility Conditions, and Applications_, Springer. ([doi:10.1007/978-3-540-78319-0_2](https://doi.org/10.1007%2f978-3-540-78319-0_2))

which I'll have a look through and then put a script together.
